enhancements/monitoring/hypershift-monitoring.md

```diff
@@ -0,0 +1,392 @@
+N/A. HyperShift is a new project.
+
+### Documentation Required
+
+User facing docs:
+
+* HyperShift client monitoring and how to configure metrics remote write (push API) to desired backend.
+
+Internal docs:
+
+* RHOBS entry-points and usage.
+
+### Open Questions
+
+### FAQ
+
+Some common questions gathered during initial proposal discussions.
+
+#### What About Layered Service / Addons Monitoring?
+
+This was discussed and solved by [Monitoring Stacks](monitoring-stack-operator.md). For HyperShift context, Addons requirements are no different to what we assume Customer would want to do / use. So in the context of this enhancement Addons are just another CUS.
+
+TBD explain risk in terms of having Addons monitoring running on "unmanaged" data-plane.
+
+#### How HyperShift Admin is any different from a Cluster Administrator?
+
+They manage multiple clusters in HyperShift topology, so multiple control planes on single management OSD. So there is added complexity on top. But technically they are administrating clusters. 
+
+#### How this can work for on-premise HyperShift?
+
+As mentioned in [Big picture](#the-big-picture), for CUS local use, metrics can be shipped to many different solutions. For example:
+
+* Data can land to locally deployed Observatorium. Obsevatorium is our open-source project that can already be installed anywhere to fulfil the proposed architecture. The first time user onboarding experience is certainly less than optimal, though a decent level of investment in better documentation, reusable dashboards and alerts and the operator deployment model would make this a drastically better experience.
+* 3rd-party vendor like Grafana Cloud, Amazon or Google Managed Prometheus, Logz.io and others.
+
+Exact offering is still to be discussed, but we need to acknowledged that managing monitoring is and expensive business.
+
+### Test Plan
+
+TBD
+
+## Implementation History
+
+If all parties will agree on the proposed direction we need heavy focus on the actions required:
+
+1. Add Alerting on top of Observatorium, then deploy it on RHOBS.
+2. Host Grafana on RHOBS.
+3. Double down on the planned performance and efficiency work for Thanos Query path.
+4. Finalize Monitoring Stack development.
+5. Acceptance test and load test.
+
+We would also propose to join forces. E.g OSD SRE who are right now helping with solving the monitoring needs could contribute directly to Observatorium, RHOBS or Monitoring Stacks.
+
+At further stage:
+
+6. Deploying multiple regional RHOBS clusters. Things that needs to be figured out:
+  a. Federation of view for those clusters
+  b. Managing more deployments
+  c. Service Discovery of those on client (HyperShift side)
+7. Provide efficient forwarding metrics collection solution (aka Prometheus agent).
+
+## Drawbacks
+
+* Different than OSD team effort to maintain critical Observability dependency with high SLO.
+* No native local storage solution
+
+## Alternatives
+
+#### Pay for Observability Vendor
```

Comment:
This is not an alternative to the proposed changes in Hypershift, this is an alternative to the in-house observability product.

Reply:
I think it's a viable alternative we already use for logging AFAIK, no?

I feel it's integral to the solution we propose.